liberty-managed/src/main/java/org/jboss/arquillian/container/was/wlp_managed_8_5/WLPManagedContainer.java

```diff
@@ -1201,6 +1218,343 @@ private void throwDeploymentException(String applicationName) throws DeploymentE
+                       log.finest("fileIsForStateChangeException:" + fileIsForStateChangeException + " " + line);
+                       
+                       fileIsForCorrectApplication = fileIsForCorrectApplication || line.contains("appName") && line.contains(appName);
+                       log.finest("fileIsForCorrectApplication:" + fileIsForCorrectApplication + "(" + appName + ")"  + " " + line);
+
+                       // Have we found all the criteria we are looking for?
+                       if (fileIsForStateChangeException && fileIsForCorrectApplication) {
+                           // We have found what we are looking for
+                           log.finest("Found StateChangeExceptionFfdc for " + appName + ": " + ffdcFile.getAbsolutePath());
+                           return ffdcFile;
+                       }
+                       
+                   }
+                   log.finest("Did not find StateChangeExceptionFfdc for " + appName + ": " + ffdcFile.getAbsolutePath() + "(file was scanned completely)");
+                   return ffdcFile;
+
+               } catch (IOException e) {
+                   log.warning("File " + ffdcFile.getAbsolutePath() + " is not readable");
+                   continue; // File has perhaps deleted but not necessarily a disaster if we find what we need later
+               }
+           }
+           
+       } finally {
+           if (br != null) {
+               try {
+                   br.close();
+               } catch (IOException e) {
+                   log.warning(e.getMessage());
+               }
+           }
+       }
+       
+       // We did not find what we are looking for
+       log.finest("No StateChangeExceptionFfdc for " + appName + " was found ");
+       return null;
+   }
+
+   /**
+    * Get the set of FFDC files, we are only called when we are expecting
+    * to see some FFDC's so we will loop, waiting for 1/100th of a second
+    * and up to 10 seconds until we see at least on FFDC. Normally we will
+    * not have to wait at all. 
+    * 
+    * @return an array of File from the FFDC dir that start "ffdc_"
+    * @throws IOException
+    */
+   private File[] getFfdcFilesSince( Long deployTime ){
+	   
+	   // We want to look for all FFDCs from at least 1 second before deploy until now...
+	   final long AT_LEAST_ONE_SECOND = (1 * 1000)+1;
+	   // Perhaps we have not passed in the deploy time - if so scan all FFDCs:
+	   final long since = deployTime!=null?(deployTime-AT_LEAST_ONE_SECOND):0;
+       File[] ffdcFiles;
+       
+       int attempts = 0;
+       do {
+    	       try {
+				Thread.sleep(100);
+			} catch (InterruptedException e1) {
+				// do nothing
+			}
+           File ffdcDir = null;
+           try {
+               ffdcDir = new File(getLogsDirectory() + "/ffdc");
+           } catch (IOException e) {
+               // TODO Auto-generated catch block
+               e.printStackTrace();
```

Comment:
We need to handle this exception. I'd be tempted to either declare that this method (and any parents) throws `IOException`, or wrap it in a `RuntimeException` and throw that.

If we don't do anything here, the following line will throw a `NullPointerException` which will be harder to diagnose.